### PR TITLE
Updates from `electron-react-boilerplate`

### PR DESCRIPTION
**Pull request by irahopkinson**
- brings up to latest `electron-react-boilerplate` **main** branch (at sha 7e0d6f9bd6a3490f418e8489760ee8f67ae4b97e).
- fixed warnings on `npm i` about resolving dependencies.
- fixed 1 security vulnerability

The first 6 commits are adaptations from the `electron-react-boilerplate` repo. The 7th commit is to fix an issue with eslint rules not coming through ERB as previously.

### CODE REVIEW

**Review by tjcouch-sil — CHANGES_REQUESTED** [review]
Thanks for the hard work on this! It's great to get dependencies updated, and I like that you replaced a lot of eslint rules with typescript-eslint rules. Just a few questions about some things.

File: src/renderer/preload.d.ts [hunk 1]
```diff
@@ -1,6 +1,7 @@
 import { ElectronAPIHandler } from 'main/preload';
 
 declare global {
+  // eslint-disable-next-line no-unused-vars
```
[the remark this answers is not in the document]
Comment (irahopkinson) [reply src/renderer/preload.d.ts]: Done. It was needed earlier when I hadn't added back in the TS rules and missed taking it back out, thanks for noticing.

File: src/shared/globalThis.ts [hunk 2]
```diff
@@ -4,7 +4,6 @@
 /**
  * Variables that are defined in global scope. These must be defined in main.ts (main), index.ts (renderer), and extension-host.ts (extension host)
  */
-
```
[the remark this answers is not in the document]
Comment (irahopkinson) [reply src/shared/globalThis.ts]: Done. Earlier this file had other changes which I reverted and missed this (useless) change. Thanks for adding it to the style thread.

File: .eslintrc.js [hunk 3]
```diff
@@ -26,20 +34,25 @@ module.exports = {
...
     'no-plusplus': ['error', { allowForLoopAfterthoughts: true }],
+    'no-use-before-define': 'off',
     'prettier/prettier': ['warn', { tabWidth: 2, trailingComma: 'all' }],
     'react/jsx-indent-props': ['warn', 2],
     'react/jsx-props-no-spreading': ['error', { custom: 'ignore' }],
     'react/require-default-props': 'off',
 
     // #endregion
   },
+  globals: {
```
[the remark this answers is not in the document]
Comment (irahopkinson) [reply .eslintrc.js:45]: It declares what globals we use and their permissions, in this case `globalThis` is readonly, i.e. we can write to its members but we can't add more members than what have been declared already.

Try taking it out and running `npm run lint` (remember to `npm i` first). It shows as undefined everywhere it is used:

image

**Comment by irahopkinson** [comment]
> ... I like that you replaced a lot of eslint rules with typescript-eslint rules.
For some of them the TS rule says to turn off the ES rule as it will give false positives. For others they can both be on without issue.

However, I think these rules where already pulled in previously via ERB (via airbnb, etc) and I wonder why ERB is no longer applying them. My concern is that there may be other rules that aren't being applied anymore but our code is compliant so we don't notice they are off. I couldn't see any issues on the ERB repo about this though.

**Review by tjcouch-sil — APPROVED** [review]

**Pull request merged**